primitives/src/pcs/univariate_kzg/mod.rs

```diff
@@ -530,19 +540,28 @@ mod tests {
     }
 
     #[test]
-    fn test_batch_open() -> Result<(), PCSError> {
+    fn test_multi_open() -> Result<(), PCSError> {
         type E = Bls12_381;
         type Fr = ark_bls12_381::Fr;
 
         let mut rng = test_rng();
-        let max_degree = 32;
+        let max_degree = 33;
         let pp = UnivariateKzgPCS::<E>::gen_srs_for_testing(&mut rng, max_degree)?;
 
         for _ in 0..5 {
-            let degree = rng.gen_range(10..33); // any degree (even non-power-of-two)
+            let degree = rng.gen_range(5..max_degree / 2); // any degree (even non-power-of-two)
             let num_points = rng.gen_range(5..max_degree);
+            ark_std::println!(
+                "Multi-opening: poly deg: {}, num of points: {}",
+                degree,
+                num_points
+            );
 
-            let (ck, _) = UnivariateKzgPCS::<E>::trim(&pp, degree, None)?;
+            // NOTE: THIS IS IMPORTANT FOR USER OF `multi_open()`!
+            // since we will pad your polynomial degree to the next_power_of_two, you will
+            // need to trim to the correct padded degree as follows:
+            let padded_degree = UnivariateKzgPCS::<E>::padded_degree_in_fk23(degree)?;
```

Comment:
This is unfortunate. Padding the degree to the next power of two is an implementation detail the user should not need to worry about. Not sure how best to handle this problem.

I wonder whether we should add a `trim_power_of_two` method to `PolynomialCommitmentScheme` trait?

Reply:
I absolutely agree with you @ggutoski, I also had this concern when I wrote the code. this is a sign of leaky abstraction.

> whether we should add a trim_power_of_two method to PolynomialCommitmentScheme trait?

probability not at the trait level since `multi_open_rou` is a `UnivariateKzgPCS`-specific function anyway, maybe:
- add `fn trim_power_of_two()` on this specific struct
- add proper error message inside `multi_open_rou()` to hint the user to use the above function during trimming.

Reply:
>     * add `fn trim_power_of_two()` on this specific struct

You mean "...on this specific *trait*" right? ie. `UnivariatePCS` trait.

Another option is to re-implement `trim()` for `UnivariateKzgPCS` to *always* trim to the next power of two, regardless of whether the caller wants it or not. In the worst case it leads to keys that are twice as large as necessary. But it does successfully hide this implementation detail from the user.

Reply:
>You mean "...on this specific trait" right? ie. UnivariatePCS trait. 

you are right. lemme re-edit my comments above.